contracts/xGNANABribes.sol

```diff
@@ -0,0 +1,155 @@
+        if(withRevert && !synced) {
+            revert("delegatee not registered");
+        }
+    }
+
+    function getRegisteredCallbackDelegateesCount() external view returns (uint256) {
+        return _registeredCallbackDelegatees.length();
+    }
+
+    function getRegisteredCallbackDelegateesAt(uint256 index) external view returns (address) {
+        return _registeredCallbackDelegatees.at(index);
+    }
+    
+    function getAllRegisteredCallbackDelegatees() external view returns (address[] memory) {
+        return _registeredCallbackDelegatees.values();
+    }
+
+    function registerCallbackDelegatee(IERC20VotesDelegatee callbackDelegatee) external override onlyRole(DELEGATEE_MANAGER_ROLE) {
+        require(callbackDelegatee.supportsInterface(type(IERC20VotesDelegatee).interfaceId), "Does not support interface");
+        if(!_registeredCallbackDelegatees.contains(address(callbackDelegatee))) {
+            _registeredCallbackDelegatees.add(address(callbackDelegatee));
+            emit CallbackDelegateRegistered(callbackDelegatee);
+        }
+    }
+
+    function removeCallbackDelegatee(IERC20VotesDelegatee callbackDelegatee) external override onlyRole(DEFAULT_ADMIN_ROLE) {
+        require(_registeredCallbackDelegatees.contains(address(callbackDelegatee)), "not registered");
+        _registeredCallbackDelegatees.remove(address(callbackDelegatee));
+        emit CallbackDelegateRemoved(callbackDelegatee);
+    }
+
+    function getVotesDelegatedTo(address delegator, address delegatee) external view override returns (uint256) {
+        address currentDelegatee = delegates(delegator);
+        if(currentDelegatee == delegatee) {
+            return balanceOf(delegator);
+        } else {
+            return 0;
+        }
+    }
+
+    /**
+     * @dev This hook is called after _mint and _burn even though
+     *  xGNANA is non-transferrable
+     */
+    function _afterTokenTransfer(
+        address from,
+        address to,
+        uint256 amount
+    ) internal override(xGNANA) {
+        super._afterTokenTransfer(from, to, amount);
+        // After balances are updated, sync external 
+        _syncDelegateVotingPower(delegates(from), from);
+        _syncDelegateVotingPower(delegates(to), to);
+    }
+
+    /**
+     * @dev Change delegation for `delegator` to `delegatee`.
+     *
+     * Sync delegation power with external delegation contract for bribes.
+     */
+    function _delegate(address delegator, address delegatee) internal override(ERC20Votes) {
+        address fromDelegate = delegates(delegator);
+        super._delegate(delegator, delegatee);
+
+        _syncDelegateVotingPower(fromDelegate, delegator);
+        _syncDelegateVotingPower(delegatee, delegator);
+    }
+
+    function _syncDelegateVotingPower(address delegator, address delegatee) internal returns (bool){
+        if(_registeredCallbackDelegatees.contains(delegatee)) {
+            IERC20VotesDelegatee(delegatee).syncDelegatePower(delegator);
```

Comment:
The tricky part is that an external contract could cause a denial of service if they revert inside this function.

The whitelist is important for that, but may not catch the edge cases. Paladin strongly recommended against using a try/catch as there is a way to purposely throw it which could allow users to unstake xGNANA and still have bribe rewards on another contract. 

The nice thing is that worst case we can de-whitelist a `delagatee` contract and it would allow xGNANA to be `redeem`ed again.